### ISSUE TITLE
plugin makes blender crash

### ISSUE THREAD

**Issue by zivoy** [chiragraman/mesh-tension#2]
**steps to reproduce**
1. open blender 
2. install the addon
3. don't have any of the `Tenshion Maps` checkboxes checked, and have `Lock Interface` unchecked
4. press f12
5. watch blender crash
5.1. it won't crash if you enable `Lock Interface`

**abbreviated console log**
```
Traceback (most recent call last):
  File "mesh_tension.py", line 648, in frame_change_pre
    handle_common_pre(scene)
  File "mesh_tension.py", line 613, in handle_common_pre
    refresh_scene_tension_objects(scene)
  File "mesh_tension.py", line 271, in refresh_scene_tension_objects
    clear_pristine_meshes()
  File "mesh_tension.py", line 205, in clear_pristine_meshes
    for bm in pristine_meshes.values():
RecursionError: maximum recursion depth exceeded while calling a Python object
Fatal Python error: _Py_CheckRecursiveCall: Cannot recover from stack overflow.
Python runtime state: initialized
```

**Comment by chiragraman** [comment]
Thanks for the issue; what version are you on? Just to confirm, this is the setup you are in when you render (disabled interface lock, disabled tension maps):

![issue-ss1](https://user-images.githubusercontent.com/1950941/126645513-bfba8c3a-7ef2-4008-8316-61d84ca89779.png)

If so, it renders fine for me:
![issue-ss2](https://user-images.githubusercontent.com/1950941/126645206-6b6e96fb-1c0e-4dfc-be00-a1b6523feada.png)

If you can give me a little more information about your blender version I can try and reproduce this. Are you testing with the default startup scene?

[Edit]: 
In general, the lock interface requirement is there to avoid issues arising from the situations described below from the official documentation. Like in the case of this issue, this shouldn't be a problem if tension maps aren't enabled, but I thought I'll leave it here in case it helps someone:

`Altering data from handlers should be done carefully. While rendering the frame_change_pre and frame_change_post handlers are called from one thread and the viewport updates from a different thread. If the handler changes data that is accessed by the viewport, this can cause a crash of Blender. In such cases, lock the interface (Render → Lock Interface or bpy.types.RenderSettings.use_lock_interface) before starting a render.`

**Comment by zivoy** [comment]
here is a video of me running version 1.5.4 of the addon on the portable version of blender 2.93.1, I get the same results as my main install which is on the same version

https://user-images.githubusercontent.com/16857861/126722071-31c96992-63d4-49ac-9e60-242031112d1e.mp4

**Comment by zivoy** [comment]
its not crashing now that I updated the plugin to ver 1.5.5

**Comment by chiragraman** [comment]
@zivoy Thanks for going to the effort of making the video, appreciate it :)
Yeah I fixed a bug in 1.5.5 that affects the frame change handlers (I had a boolean flipped). From your initial console log I thought the error was somewhere else so didn't mention it earlier.
